src/plugins/vis_type_xy/public/vis_component.tsx

```diff
@@ -245,21 +245,58 @@ const VisComponent = (props: VisComponentProps) => {
   const isDarkMode = getThemeService().useDarkMode();
   const getSeriesName = getSeriesNameFn(config.aspects, config.aspects.y.length > 1);
 
+  const splitAccessors = config.aspects.series?.map(({ accessor, formatter }) => {
+    return { accessor, formatter };
+  });
+
+  const allSeries = useMemo(() => getAllSeries(visData.rows, splitAccessors, config.aspects.y), [
+    config.aspects.y,
+    splitAccessors,
+    visData.rows,
+  ]);
+
   const getSeriesColor = useCallback(
     (series: XYChartSeriesIdentifier) => {
-      const seriesName = getSeriesName(series);
+      const seriesName = getSeriesName(series) as string;
```

Comment:
Do you need this type casting?

```suggestion
      const seriesName = getSeriesName(series);
```

Reply:
Yes because it fails some lines later on `overwrite colors`.